Add instructions and operators for new bytecode format of PyText model

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33555 Add instructions and operators for new bytecode format of PyText model**

A quick fix for the PyText model (in internal production) on the new bytecode format.

Differential Revision: [D20008266](https://our.internmc.facebook.com/intern/diff/D20008266)

Commit:
Add instructions and operators for new bytecode format of PyText model

Commit:
Update on "Add instructions and operators for new bytecode format of PyText model"

dreiss commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -190,5 +190,38 @@ void testLiteInterpreterLoadOrigJit() {
   ASSERT_THROWS_WITH(_load_for_mobile(ss), "file not found");
 }
 
+//void testLiteInterpreterPytext() {
```
Please uncomment or remove this.

dreiss commented on torch/csrc/jit/mobile/register_mobile_ops.cpp:
```diff
@@ -132,6 +133,37 @@ void warn_kernel(const c10::OperatorHandle& op, Stack* stack) {
+     tuple->elements().end());
+}
+
+void format_kernel(const c10::OperatorHandle& op, Stack* stack) {
```
This looks duplicated from full jit.  Any way to reuse code?

zdevito replied:
This can just call formatFunc from vararg_functions.h

iseeyuan replied:
vararg_functions.cpp may not be included in the BUCK target for internal build. Since this file will be removed soon, I may just land it for now to quickly resolve the internal test issue, and remove them together with the file.

zdevito commented on torch/csrc/jit/mobile/register_mobile_ops.cpp:
```diff
@@ -132,6 +133,37 @@ void warn_kernel(const c10::OperatorHandle& op, Stack* stack) {
   pop(*stack);
 }
 
+void tupleunpack_kernel(const c10::OperatorHandle& op, Stack* stack) {
```
this is `tupleUnpackFunc` from vararg_functions

zdevito reviewed (approved):
As a short-term fix, this is good. See hints for code reuse. Long term it is now possible to move operator.h into the mobile build (it no longer depends on ir), and to move the operators like tuple unpack and format that are used in both interpreters to a file included by both.

Commit:
Update on "Add instructions and operators for new bytecode format of PyText model"


A quick fix for the PyText model (in internal production) on the new bytecode format.

Differential Revision: [D20008266](https://our.internmc.facebook.com/intern/diff/D20008266)

Commit:
Update on "Add instructions and operators for new bytecode format of PyText model"


A quick fix for the PyText model (in internal production) on the new bytecode format.

Differential Revision: [D20008266](https://our.internmc.facebook.com/intern/diff/D20008266)
